### ISSUE TITLE
[Screen reader] Trees should have aria-labels or aria-labelledby attributes

### ISSUE THREAD

**Issue by microbit-robert** [RaspberryPiFoundation/blockly#9464]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

The toolbox tree and flyout tree are missing these attributes which I think are required to follow best practice. For most screen readers you can get enough context from the region change, but for some reason, JAWS does not announce the region change and the trees themselves don’t have labels. I have tried changing the JAWS settings (general verbosity and web page specific verbosity) to no avail. This change makes the NVDA readout more verbose, but I think this is necessary to support anyone using JAWS. 

It’s particularly bad in MakeCode without the tree labels as we move from the Basic category to the Basic flyout where the first item is the Basic `FlyoutLabel`, which is very confusing.

### Reproduction steps

1. 
2. 
3.

### Stack trace

```shell

```

**Comment by BenHenning** [comment]
I think I'm not opposed by this, particularly with the use case you mentioned, but I'm not sure what labels we should be using here to avoid redundancy. If the regions are ignored then obviously we could reuse 'Toolbox' and 'Flyout' (pending a better term) labels, but if the regions _are_ read then it will read that twice.

**Comment by BenHenning** [comment]
Actually, at least on ChromeVox the combined version isn't that bad because there's a bit of a delay between the two (reading the tree and reading the region), so the redundancy sort of makes sense. Going to through up a PR to see if this addresses the issue for you @microbit-robert.

**Comment by BenHenning** [comment]
Fixed in #9483.
